src/app/services/articles.service.ts

```diff
@@ -1,34 +1,42 @@
 import { Injectable } from '@angular/core';
-import { Observable, ReplaySubject } from "rxjs";
+import { Observable, Subject } from "rxjs";
+import { switchMap } from "rxjs/operators";
+import { HttpClient } from "@angular/common/http";
 
-import { articles } from '../mocks/articles.mock';
-import { Article, ArticleWithoutId } from "../models/article.model";
+import { Article } from "../models/article.model";
+import { UserService } from "./user.service";
+import { User } from "../models/user.model";
 
 @Injectable({
   providedIn: 'root'
 })
 export class ArticlesService {
-  private articles: Article[] = articles;
-  private articles$ = new ReplaySubject<Article[]>();
+  constructor(private http: HttpClient, private userService: UserService) {}
+
+  public articlesUpdated$ = new Subject<Article[]>();
+  private url = 'http://localhost:3000/api/posts';
 
   getArticles(): Observable<Article[]> {
-    this.articles$.next(this.articles);
-    return this.articles$;
+    return this.http.get<Article[]>(this.url);
+  }
+
+  addArticle(article: Article): Observable<Article> {
+    return this.userService.getUser().pipe(
```

Comment:
look's like user could not be changed in that time - so u can simply call getUser in the constructor and save current user id in param